zerver/lib/upload.py

```diff
@@ -149,6 +149,11 @@ def get_file_info(request, user_file):
     # type: (HttpRequest, File) -> Tuple[text_type, Optional[text_type]]
 
     uploaded_file_name = user_file.name
+    if not isinstance(user_file.name, str):
+        # This check makes sure that the type of the file name is not
+        # Python 2 Unicode.
+        uploaded_file_name = uploaded_file_name.encode('ascii')
```

Comment:
Yeah, it doesn't surprise me that the non-string value is coming from outside of Zulip, but I was thinking we might want to do the type conversion and/or validation in the view function, not the library.  What client did you discover this problem with?  Was it the normal Zulip browser client?